### PR TITLE
Simplify GCC compilation test

### PR DESCRIPTION
**Pull request by asmorodskyi**
Old version of tests uses  unreasonably heavy sources to compile
which took on ARM up to 2 hours to compile. This was replaced with
light-weight gawk sources which contains both C and C++ sources.

### CODE REVIEW

**Comment by asmorodskyi** [comment]
proof run -  http://kimball.arch.suse.de/tests/398

**Comment by asmorodskyi** [comment]
@mnowaksuse  , @sysrich  , @okurz  please take a look. 

And of course I will fix tidy thingy :)

**Comment by coolo** [comment]
I think we should bootstrap gentoo :)

**Comment by coolo** [comment]
wrong button :)

**Comment by okurz** [comment]
+1

File: tests/toolchain/gcc_compilation.pm [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
```
Comment (mnowaksuse) [inline tests/toolchain/gcc_compilation.pm]: Maybe we don't have to drag Bernhard's copyright to this new file?

File: tests/toolchain/gcc_compilation.pm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# Copyright © 2012-2017 SUSE LLC
```
Comment (mnowaksuse) [inline tests/toolchain/gcc_compilation.pm]: `2017`-only, maybe?

File: tests/toolchain/gcc_compilation.pm [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: C toolchain module test
+# Maintainer: Michal Nowak <mnowak@suse.com>
```
Comment (mnowaksuse) [inline tests/toolchain/gcc_compilation.pm]: No need to be humble, you are the maintainer now :).

File: tests/toolchain/gcc_compilation.pm [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: C toolchain module test
```
Comment (mnowaksuse) [inline tests/toolchain/gcc_compilation.pm]: In git commit message you claim it tests both C and C++ compilers, make it clear in the test module as well, please.

**Comment by mnowaksuse** [comment]
@asmorodskyi Did you have chance to actually test it on non-x86_64 archs?

File: tests/toolchain/gcc_compilation.pm (hunk 1 above)
Comment (asmorodskyi) [inline tests/toolchain/gcc_compilation.pm]: you right I am just blindy copy/paste this part :) will change everything with my mind turned on :)

**Comment by asmorodskyi** [comment]
No. you right here too . will figure out something to do proof run on ARM . will it be enough ? or you think it will be right to run on power and s390 too before merge ?

File: tests/toolchain/gcc_compilation.pm [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: C and C++ source compilation to check that gcc from toolchain module
+# Maintainer: Anton Smorodskyi <asmorodskyi@suse.com> , Michal Nowak <mnowak@suse.com>
```
Comment (asmorodskyi) [inline tests/toolchain/gcc_compilation.pm]: @mnowaksuse  if you don't mind  you would like to leave it like this :) as you still have a lot of expertise in this area

**Comment by mnowaksuse** [comment]
@asmorodskyi I prefer it to be proven on non-x86_64 archs before we merge & wonder.

File: tests/toolchain/gcc_compilation.pm (hunk 5 above)
Comment (mnowaksuse) [inline tests/toolchain/gcc_compilation.pm]: Np, with co-maintaining it.

**Comment by asmorodskyi** [comment]
let's say it will take some time :) . but I agree

**Comment by asmorodskyi** [comment]
s390 proof - http://kimball.arch.suse.de/tests/410
power is on the way . regarding ARM I discussed this with @sysrich and his advice was due to lack of ARM workers just merge PR without ARM proof run

**Comment by asmorodskyi** [comment]
power proof run http://furud.arch.suse.de/tests/40

**Pull request merged**